src/client/serialize/daos_serialize.c

```diff
@@ -1175,3 +1244,1818 @@ daos_cont_deserialize_attrs(char *filename, uint64_t *num_attrs,
+	return rc;
+}
+
+static int
+serialize_recx_array(struct hdf5_args *hdf5, daos_key_t *dkey, daos_key_t *akey, char *rec_name,
+		     int *akey_index, daos_handle_t *oh, daos_iod_t *iod, uint64_t *bytes_read)
+{
+	int			rc = 0;
+	hid_t			status = 0;
+	int			i = 0;
+	int			attr_num = 0;
+	int			buf_len = 0;
+	int			path_len = 0;
+	uint32_t		number = 5;
+	size_t			nalloc = 0;
+	daos_anchor_t		recx_anchor = {0}; 
+	daos_anchor_t		fetch_anchor = {0}; 
+	daos_epoch_range_t	eprs[5] = {0};
+	daos_recx_t		recxs[5] = {0};
+	daos_size_t		size = 0;
+	char			attr_name[ATTR_NAME_LEN] = {0};
+	char			attr_num_str[ATTR_NAME_LEN] = {0};
+	unsigned char		*encode_buf = NULL;
+	d_sg_list_t		sgl = {0};
+	d_iov_t			iov = {0};
+	char			*buf = NULL;
+	herr_t			err = 0;
+	hid_t			rx_dset = -1;
+	hid_t			selection_attr = -1;
+	hid_t			rx_dtype = -1;
+	hid_t			rx_dspace = -1;
+	hid_t			rx_memspace = -1;
+	hid_t			attr_dspace = -1;
+	hid_t			attr_dtype = -1;
+	hid_t			plist = -1;
+	hsize_t			rx_dims[1];
+	hsize_t			rx_max_dims[1];
+	hsize_t			rx_chunk_dims[1];
+	hsize_t			mem_dims[1];
+	hsize_t			attr_dims[1];
+	hsize_t			start;
+	hsize_t			count;
+
+	rx_dims[0] = 0;
+	rx_max_dims[0] = H5S_UNLIMITED;
+
+	/* TODO: consider other chunk sizes or possibly use different
+	 * chunk sizes for different dkeys/akeys. */
+	rx_chunk_dims[0] = 1024;
+
+	plist = H5Pcreate(H5P_DATASET_CREATE);
+	if (plist < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed property list "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+
+	rx_dspace = H5Screate_simple(1, rx_dims, rx_max_dims);
+	if (rx_dspace < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to create rx_dspace "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+
+	err = H5Pset_layout(plist, H5D_CHUNKED);
+	if (err < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to create rx_dspace "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	
+	err = H5Pset_chunk(plist, 1, rx_chunk_dims);
+	if (err < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to create rx_dspace "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	
+	/* need to do a fetch for size, so that we can
+	 * create the dataset with the correct datatype size */
+	number = 1;
+	rc = daos_obj_list_recx(*oh, DAOS_TX_NONE, dkey, akey, &size, &number, NULL, eprs,
+				&fetch_anchor, true, NULL);
+	if (rc != 0) {
+		D_ERROR("Failed to list recx "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	if (number == 0) {
+		rc = 0;
+		D_GOTO(out, rc);
+	}
+
+	if (size > 2000) {
+		D_ERROR("recx size is too large: %d, "DF_RC"\n", (int)size, DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+
+	/* create the dataset with the correct type size */
+	rx_dtype = H5Tcreate(H5T_OPAQUE, size);
+	if (rx_dtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed create rx_dtype "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+
+	rx_dset = H5Dcreate(hdf5->file, rec_name, rx_dtype, rx_dspace, H5P_DEFAULT,
+			    plist, H5P_DEFAULT);
+	if (rx_dset < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed create rx_dset "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+ 	}
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs